R/get_coding_ssm_status.R

```diff
@@ -2,30 +2,69 @@
 #'
 #' @description Tabulate mutation status (SSM) for a set of genes.
 #'
-#' @details This function takes a vector of gene symbols and subsets the incoming MAF to specified genes. If no genes are provided, the function will default to all lymphoma genes.
-#' The function can accept a wide range of incoming MAFs. For example, the user can call this function with `these_samples_metadata` (preferably a metadata table that has been subset to the sample IDs of interest).
-#' If this parameter is not called, the function will default to all samples available with [GAMBLR.results::get_gambl_metadata]. The user can also provide a path to a MAF, or MAF-like file with `maf_path`,
```

Comment:
Suggest rephrasing so the details are more clear about the purpose:


This function takes a data frame (in MAF-like format) and converts it to a binary one-hot encoded matrix of mutation status for either a set of user-specified genes (via `gene_symbols`) or, if no genes are provided, default to all lymphoma genes. The default behaviour is to assign each gene/sample_id combination as mutated only if there is a protein coding mutation for that sample in the MAF but this can be configured to use synonymous variants in some or all genes.